### PR TITLE
Implement improved spacing for the thread list and timeline

### PR DESCRIPTION
**Pull request by luixxiul**
Closes https://github.com/vector-im/element-web/issues/21759

Steps to check the implementation:

1. Create a test room
2. Create a test user
3. Click the thread icon on the top toolbar
4. Create a thread in the test room
5. Send messages
6. Send messages from the test user's account as well

after1

With the scroll bar:
after

Without the scroll bar:
after2

type: enhancement

---
Here's what your changelog entry will look like:

## ✨ Features
 * Implement improved spacing for the thread list and timeline ([\#8337](https://github.com/matrix-org/matrix-react-sdk/pull/8337)). Fixes vector-im/element-web#21759. Contributed by @luixxiul.

Preview: https://pr8337--matrix-react-sdk.netlify.app
⚠️ Do you trust the author of this PR? Maybe this build will steal your keys or give you malware. Exercise caution. Use test accounts.

### CODE REVIEW

**Comment by luixxiul** [comment]
The unread indicator is yet to be styled (I am not sure how to have it displayed) :frowning:

**Comment by luixxiul** [comment]
The existing visual bug that the top right and bottom border-radius inside the thread panel are not effective should be able to be fixed by merging the style rules with those of the chat panel with a maximized widget (confirmed locally).

after

Since this is a different issue and requires some refactoring, I guess this should be addressed with another PR, but it can be fixed on this PR if that is preferred.

**Review by t3chguy — APPROVED** [review]
Looks sane code-wise!

**Comment by luixxiul** [comment]
Thank you for reviewing!

**Comment by luixxiul** [comment]
The conflict was fixed.

**Comment by luixxiul** [comment]
https://github.com/matrix-org/matrix-react-sdk/pull/8337/commits/72949b322ec154e10e0f32f843851e628115368a should have not included in this PR. I'm dropping it.

**Review by janogarcia — CHANGES_REQUESTED** [review]
@luixxiul Thanks for the awesome contribution, Suguru. 👏

Just a few details that would need to be improved to get this merged.

---

**Thread list: Spacing**

I've annotated the spacing values that should be fixed (current value as measured vs expected value in respective callout).

spacing-thread-list

---

**Thread list: Contextual actions hover behavior**

Please notice how the preceding thread gets the hover focus when the user tries to reach the contextual actions buttons in the thread immediately below it.

hover-stack

---

**Thread list: Unread indicator**

> The unread indicator is yet to be styled (I am not sure how to have it displayed) 😦

You'll need two separate accounts to be able to test this (i.e., replying to your threads with a different account). That said, there's currently a bug that is preventing this from working correctly ([I reported it today](https://github.com/vector-im/element-web/issues/21927)).

**Comment by janogarcia** [comment]
@luixxiul In case you missed it (I guess GitHub doesn't notify you on edits of review comments) I added a new section in my comment for the **Thread list: Unread indicator**.

**Comment by janogarcia** [comment]
> @janogarcia Thanks for checking in detail! I'll work on the improvement per your suggestions.

@luixxiul Thank you for your time. 🙏

**Comment by janogarcia** [comment]
@t3chguy I couldn't test @luixxiul latest changes on Netlify. GitHub seems to be complaining about running workflows for first-time contributors. Can we please approve running them?

**Comment by janogarcia** [comment]
Thanks, @luixxiul!

Unfortunately, I'm not getting the same output as in your screenshots. It seems that a number of spacing regressions were introduced at some point.

---

**Thread list**

spacing

**Comment by luixxiul** [comment]
@janogarcia I'm not sure how that happened but it might be related to the conflict due to the latest change. I'll fix the conflict and ping you then..

**Comment by luixxiul** [comment]
@janogarcia The conflicts were fixed, and the spacing style rules were adjusted. Would you please check again? I've tried my best but it is hard to see 1px difference on my monitor :eyeglasses:

**Comment by janogarcia** [comment]
@luixxiul Everything's looking fine. Great job! Only if we could update `.mx_ThreadView .mx_EventTile:not([data-layout=bubble])` top padding to be `16px` ? 🙏

**Review by janogarcia — CHANGES_REQUESTED** [review]
Minimal change request in my previous comment, otherwise looking great. 👍

**Comment by luixxiul** [comment]
@janogarcia No problem, I'm going to check it very soon (I hope there is not a hacky solution required :grin:  )

**Comment by luixxiul** [comment]
I found a regression due to a recent commit of this PR, and I'm going to address it as well.

File: res/css/views/rooms/_EventTile.scss [hunk 1]
```diff
@@ -858,10 +858,10 @@ $eventTileZIndex: 10;
             padding-left: 0;
             padding-right: 0;
         }
-    }
 
-    .mx_EventTile:not([data-layout=bubble]) {
-        padding-top: 14px; // due to layout differences, this odd number matches the 18px padding-top of main tl events
+        &:not([data-layout=bubble]) {
+            padding-top: $spacing-16;
```
Comment (luixxiul) [inline res/css/views/rooms/_EventTile.scss:845]: @janogarcia `.mx_ThreadView .mx_EventTile:not([data-layout=bubble])`'s top padding is set with this line.

File: res/css/views/right_panel/_ThreadPanel.scss [hunk 2]
```diff
@@ -125,18 +126,6 @@ limitations under the License.
         }
     }
 
-    // chat panel with a maximized widget
-    &.mx_TimelineCard {
-        .mx_GroupLayout {
-            .mx_EventTile {
-                .mx_MessageTimestamp {
-                    position: absolute;
-                    width: $MessageTimestamp_width;
```
Comment (luixxiul) [inline res/css/views/right_panel/_ThreadPanel.scss]: These are partial revert of https://github.com/matrix-org/matrix-react-sdk/pull/8337/commits/addd6a723f2a49192c26dc4e2397de9157962646. They are not required thanks to the original properties on [_GroupLayout.scss](https://github.com/matrix-org/matrix-react-sdk/blob/472222c195d5d953d673519f7ba8e00cf56fc4ca/res/css/views/rooms/_GroupLayout.scss#L32).

**Comment by luixxiul** [comment]
@janogarcia thank you very much for taking your time! :smile:

File: src/components/views/rooms/EventTile.tsx [hunk 3]
```diff
@@ -1032,8 +1032,10 @@ export class UnwrappedEventTile extends React.Component<IProps, IState> {
         if (this.context.timelineRenderingType === TimelineRenderingType.Notification) {
             avatarSize = 24;
             needsSenderProfile = true;
-        } else if (this.context.timelineRenderingType === TimelineRenderingType.ThreadsList) {
-            avatarSize = 36;
+        } else if (this.context.timelineRenderingType === TimelineRenderingType.ThreadsList ||
+                   this.context.timelineRenderingType === TimelineRenderingType.Thread && !this.props.continuation
+        ) {
```
Comment (t3chguy) [inline src/components/views/rooms/EventTile.tsx:1037]:
```suggestion
        } else if (this.context.timelineRenderingType === TimelineRenderingType.ThreadsList ||
            (this.context.timelineRenderingType === TimelineRenderingType.Thread && !this.props.continuation)
        ) {
```

**Comment by luixxiul** [comment]
Going to fix the conflict

**Review by t3chguy — APPROVED** [review]
:shipit: Thanks for this!!

**Pull request merged**